models/classes/TestCategoryPreset.php

```diff
@@ -104,47 +111,55 @@ public function __construct($id, $label, $qtiCategory, $data)
         if (isset($data['featureFlag'])) {
             $this->featureFlag = (string) $data['featureFlag'];
         }
+        if (isset($data['altCategories'])) {
+            $this->altCategories = array_map('strval', $data['altCategories']);
+        }
     }
 
-    public function getId()
+    public function getId(): string
     {
         return $this->id;
     }
 
-    public function getLabel()
+    public function getLabel(): string
     {
         return $this->label;
     }
 
-    public function getQtiCategory()
+    public function getQtiCategory(): string
     {
         return $this->qtiCategory;
     }
 
-    public function getDescription()
+    public function getAltCategory(): array
+    {
+        return $this->altCategories;
+    }
+
+    public function getDescription(): ?string
     {
         return $this->description;
     }
 
-    public function getOrder()
+    public function getOrder(): int
     {
         return $this->order;
     }
 
-    public function getPluginId()
+    public function getPluginId(): string
     {
         return $this->pluginId;
     }
 
-    public function getFeatureFlag()
+    public function getFeatureFlag(): string
```

Comment:
@jsconan as far as I see in the constructor this is optional, so it needs to be a bit different https://github.com/oat-sa/extension-tao-testqti/pull/2461/files#diff-c5577ec1fd5d99becf2d82ed6ffc21e5f0cd54abfe2be7f978c690139257a473R111

```suggestion
    public function getFeatureFlag(): ?string
```

Reply:
Yes, I saw it, but the type is forced to string so that it always return with a string. Or do you think I should change the return instruction as well? If possible I'd prefer to not change too much the code I was not supposed to touch unless the change is harmless, and for this one I don't now the impact.

Reply:
IMO if it is optional, it must be null in case of not provided. It helps on validation and it is more consistent IMO.